### PR TITLE
SOC-3193 Followed post list

### PR DESCRIPTION
**Pull request by latata**
## Links

* https://wikia-inc.atlassian.net/browse/SOC-3193

## Description

Followed posts lists

## Reviewers

@Wikia/social-frontend

### CODE REVIEW

File: front/main/app/components/discussion-header.js [hunk 1]
```diff
@@ -19,11 +19,16 @@ export default Ember.Component.extend(
...
 
 		discussionSort: Ember.inject.service(),
-		isFilterApplied: Ember.computed('discussionSort.sortTypes.@each.active', 'categories.@each.selected', function () {
-			return this.get('discussionSort.sortTypes.0.active') === false ||
-				!this.get('categories').isEvery('selected', false) ||
-				this.get('discussionSort.onlyReported', true);
-		}),
+		isFilterApplied: Ember.computed('discussionSort.sortTypes.@each.active', 'categories.@each.selected',
+			'isFilterDisabled', function () {
+				if (this.get('isFilterDisabled')) {
```
Comment (mleczey) [inline front/main/app/components/discussion-header.js]: Could this be combined with below conditions? If this is not readable maybe extracting methods would help?

File: front/main/app/routes/discussion/follow.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+			});
+		},
+
+		afterModel(model, transition) {
+			this._super(...arguments);
+
+			this.goToFirstPageIfNoPosts(model, transition.queryParams);
+		},
+
+		serializeQueryParam(value, urlKey, defaultValueType) {
```
Comment (mleczey) [inline front/main/app/routes/discussion/follow.js]: Do we need to serialize/deserialize query params on following?

File: front/main/app/routes/discussion/follow.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+import DiscussionContributionRouteMixin from '../../mixins/discussion-contribution-route';
+import DiscussionFollowedPostsModel from '../../models/discussion/follow';
+import DiscussionModerationRouteMixin from '../../mixins/discussion-moderation-route';
+import DiscussionForumActionsRouteMixin from '../../mixins/discussion-forum-actions-route';
+import DiscussionModalDialogMixin from '../../mixins/discussion-modal-dialog';
+import localStorageConnector from '../../utils/local-storage-connector';
+
+const {inject} = Ember;
+
+export default DiscussionBaseRoute.extend(
```
Comment (mleczey) [inline front/main/app/routes/discussion/follow.js]: Maybe some code could be shared between following and forum?

File: front/main/app/routes/discussion/follow.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+	DiscussionForumActionsRouteMixin,
+	DiscussionModalDialogMixin,
+	{
+		queryParams: {
+			page: {
+				refreshModel: false
+			}
+		},
+
+		canModerate: null,
```
Comment (mleczey) [inline front/main/app/routes/discussion/follow.js]: Isn't this always false for following?

File: front/main/app/styles/component/discussion/_discussion-forum-tabs.scss [hunk 5]
```diff
@@ -18,18 +18,15 @@
...
-		li {
-			&.sort-trending {
-				border-bottom-color: $discussion-sort-inactive-color;
-				color: $discussion-sort-inactive-color;
-				pointer-events: none;
-			}
-		}
+@media #{$mobile-range} {
+	.discussion-forum-tabs {
+		width: 100%;
```
Comment (mleczey) [inline front/main/app/styles/component/discussion/_discussion-forum-tabs.scss]: Please keep alphabetic order.

File: front/main/app/styles/component/discussion/_discussion-header.scss [hunk 6]
```diff
@@ -88,14 +88,19 @@ $site-header-height: 50px;
 		text-align: right;
 		top: 0;
 
+		&.disabled {
+			cursor: default;
+			.filter-default {
```
Comment (mleczey) [inline front/main/app/styles/component/discussion/_discussion-header.scss]: Please add new line here.

File: front/main/app/styles/component/discussion/_discussion-no-followed-posts.scss [hunk 7]
```diff
@@ -0,0 +1,34 @@
+.no-followed-posts {
```
Comment (mleczey) [inline front/main/app/styles/component/discussion/_discussion-no-followed-posts.scss]: Please keep alphabetic order of css properties.

File: front/main/app/styles/component/discussion/_discussion-no-followed-posts.scss [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+	margin: 30px auto;
+	text-align: center;
+	color: #666;
```
Comment (mleczey) [inline front/main/app/styles/component/discussion/_discussion-no-followed-posts.scss]: Is this color general or specific to wiki? Could you create variable or reuse variable in settings.scss?

File: front/main/app/styles/component/discussion/_discussion-no-followed-posts.scss [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+	width: 272px;
+
+	.header {
+		font-weight: $font-weight-bold;
+		color: #333;
```
Comment (mleczey) [inline front/main/app/styles/component/discussion/_discussion-no-followed-posts.scss]: Please see comment above.

File: front/main/app/styles/component/discussion/_discussion-no-followed-posts.scss [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+
+	.follow-xl-icon {
+		height: 42px;
+		width: 32px;
+		margin-bottom: 20px;
+	}
+
+	.find-posts-button {
+		display: block;
+		border-style: solid;
```
Comment (mleczey) [inline front/main/app/styles/component/discussion/_discussion-no-followed-posts.scss]: Border property can be combined.

File: front/main/app/styles/component/discussion/_discussion-no-followed-posts.scss [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+		border-width: 1px;
+		font-size: 0.75rem;
```
Comment (mleczey) [inline front/main/app/styles/component/discussion/_discussion-no-followed-posts.scss]: Can you reuse some variables here? (`font-size`, `font-weight`, letter-spacing`, `line-height`) If there is no such a variable maybe you could create it here? (Or if this is only specific to this element leave it as it is.

File: front/main/app/styles/layout/_discussion-items-list.scss [hunk 12]
```diff
@@ -36,6 +36,10 @@ $discussion-lists-post-margin-bottom: 10px;
 	}
 }
 
+.discussion.forum.discussion-reported-posts {
```
Comment (mleczey) [inline front/main/app/styles/layout/_discussion-items-list.scss]: Please use new line after each css class. So this should have three levels of nesting.

File: front/main/app/templates/components/discussion-forum-tabs.hbs [hunk 13]
```diff
@@ -0,0 +1,13 @@
+{{link-to
+		(i18n 'main.tab-all' ns='discussion')
+		'discussion.forum'
+		tagName="li"
```
Comment (mleczey) [inline front/main/app/templates/components/discussion-forum-tabs.hbs]: Please keep `''` convention.

File: front/main/app/templates/components/discussion-forum-tabs.hbs [hunk 14]
```diff
@@ -0,0 +1,13 @@
...
+		class=(if allTabActive 'active-element-theme-color active-element-border-theme-color' 'not-active')
```
Comment (mleczey) [inline front/main/app/templates/components/discussion-forum-tabs.hbs]: Do you need `not-active` class here?

File: front/main/app/templates/components/discussion-forum-tabs.hbs [hunk 15]
```diff
@@ -0,0 +1,13 @@
...
+}}
+{{link-to
+		(i18n 'main.tab-following' ns='discussion')
+		'discussion.follow'
+		tagName='li'
+		class=(if followTabActive 'active-element-theme-color active-element-border-theme-color' 'not-active')
```
Comment (mleczey) [inline front/main/app/templates/components/discussion-forum-tabs.hbs]: Same here.

File: front/main/app/templates/components/discussion-forum-wrapper.hbs [hunk 16]
```diff
@@ -42,9 +42,13 @@
 		upvote=(action this.attrs.upvote)
 	}}
 {{else}}
-	{{discussion-empty-forum-message
-		setEditorActive=(action this.attrs.setEditorActive)
-	}}
+	{{#if isFollewedPostsView }}
```
Comment (mleczey) [inline front/main/app/templates/components/discussion-forum-wrapper.hbs]: Is it possible to move this to follow tab - to more specific place?

File: front/main/app/templates/components/discussion-forum-tabs.hbs (hunk 14 above)
Comment (latata) [inline front/main/app/templates/components/discussion-forum-tabs.hbs]: Yes, if there is no not-active class there are problems with priority of styles and active colors are not applied.

File: front/main/app/templates/components/discussion-forum-tabs.hbs (hunk 15 above)
Comment (latata) [inline front/main/app/templates/components/discussion-forum-tabs.hbs]: look at my answer above

File: front/main/app/routes/discussion/follow.js (hunk 3 above)
Comment (latata) [inline front/main/app/routes/discussion/follow.js]: I think there is no so much to share.

**Pull request merged**